core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -394,53 +396,68 @@ private long checkDeletedAndGCed(VersionValue versionValue) {
     }
 
     @Override
-    public boolean index(Index index) {
-        final boolean created;
+    public void index(Index index) {
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             if (index.origin().isRecovery()) {
                 // Don't throttle recovery operations
-                created = innerIndex(index);
+                innerIndex(index);
             } else {
                 try (Releasable r = throttle.acquireThrottle()) {
-                    created = innerIndex(index);
+                    innerIndex(index);
                 }
             }
-        } catch (IllegalStateException | IOException e) {
-            try {
-                maybeFailEngine("index", e);
-            } catch (Exception inner) {
-                e.addSuppressed(inner);
+        } catch (OutOfMemoryError | IllegalStateException t) {
```

Comment:
@bleskes The conversation belongs here because we are not catching it in master (I already removed it in #19231) and I noticed a regression slip in here. :smile: